tests/components/ring/test_sensor.py

```diff
@@ -72,6 +72,9 @@ def test_sensor(self, mock):
         for device in self.DEVICES:
             device.update()
             if device.name == 'Front Battery':
+                expected_icon = icon_for_battery_level(
+                    battery_level=int(device.state), charging=False)
+                assert device.icon == expected_icon
```

Comment:
In a future PR these tests should be rewritten to not interact directly with the entity. Instead we should set up the component and platform, let it add an entity, then get the state of the entity from the core state machine and assert the state. The state is the truth of the entity.

Not depending on the entity in tests will make them more robust as the entity implementation can be changed without changing the tests as long as the state remains the same.

We also want tests to be written as standalone pytest test functions, not test methods, as part of a test class.

Reply:
Ace. Can you suggest a component that does this correctly and I'll see if I can figure it out for my next pr

Reply:
The heos tests are good. Remove the config entry parts and add use of the available pytest fixture `requests_mock` to mock requests.

https://github.com/home-assistant/home-assistant/blob/dev/tests/components/heos/test_media_player.py